### PR TITLE
Fix pydantic dataclass problem with dataclasses.field default

### PR DESCRIPTION
**Pull request by hramezani**
Fixes https://github.com/pydantic/pydantic/issues/7895

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by hramezani** [comment]
Please review

File: pydantic/_internal/_fields.py [hunk 1]
```diff
@@ -248,7 +248,11 @@ def collect_dataclass_fields(
         if is_classvar(ann_type):
             continue
 
-        if not dataclass_field.init and dataclass_field.default_factory == dataclasses.MISSING:
+        if (
+            not dataclass_field.init
+            and dataclass_field.default_factory == dataclasses.MISSING
+            and dataclass_field.default == dataclasses.MISSING
```
Comment (PrettyWood) [inline pydantic/_internal/_fields.py:254]: I would have swapped the order because `default` is used more often that `default_factory`

**Review by PrettyWood — APPROVED** [review]
LGTM! Thank you

File: pydantic/_internal/_fields.py (hunk 1 above)
Comment (hramezani) [reply pydantic/_internal/_fields.py]: Sure! Done

**Review by PrettyWood — APPROVED** [review]

**Pull request merged**